src/azure-cli/azure/cli/command_modules/acr/import.py

```diff
@@ -95,45 +96,44 @@ def acr_import(cmd,
                                               target_tags=target_tags,
                                               untagged_target_repositories=repository,
                                               mode=ImportMode.force.value if force else ImportMode.no_force.value)
+    
+    try:
+        if no_wait:
+            return sdk_no_wait(no_wait, client.import_image, resource_group_name, registry_name, import_parameters)
+        else:
+            result_poller = client.import_image(
+                resource_group_name=resource_group_name,
+                registry_name=registry_name,
+                parameters=import_parameters)
 
-    result_poller = client.import_image(
-        resource_group_name=resource_group_name,
-        registry_name=registry_name,
-        parameters=import_parameters)
-
-    if no_wait:
-        return result_poller
+            return LongRunningOperation(cmd.cli_ctx, 'Importing image...')(result_poller)
 
-    return _handle_result(cmd, result_poller, source_registry, source_image, registry)
+    except CLIError as e:
+        _handle_result(e, cmd, source_registry, source_image, registry)
 
 
-def _handle_result(cmd, result_poller, source_registry, source_image, registry):
+def _handle_result(e, cmd, source_registry, source_image, registry):
```

Comment:
@CarolineNB nit: since this class is now an exception handler only, we can rename it to `_handle_exception`.

Reply:
agreed. Maybe name it `_handle_import_exception`